### ISSUE TITLE
UnhandledPromiseRejectionWarning (node) / Uncaught (in promise) (browser) when passing text as undefined 

### ISSUE THREAD

**Issue by gytisgreitai** [gitbrent/PptxGenJS#678]
node code:
```
  try {
    slide.addText([{text: undefined, options: {breakLine: true}}], {x:1, y:1, w: 1, h:1})
  } catch (e) {
    console.log('this will not be invoked')
    console.error(e)
  }
```

Fails with:
```
(node:63649) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
```
Example to reproduce in browser:

Fiddle: https://jsfiddle.net/6ayzoxh2/

Clearly this is an error on my side, and I should not pass the undefined text, but it would be cool if your code would throw, then I could handle it

**Comment by ankon** [comment]
We're getting hit by this as well.

The problem indeed is that one cannot even catch this exception at all, so all our "defensive coding" didn't help.

I'm right now looking into this, and a very initial hunch is that the whole `save` code is failing to properly _reject_ the involved promises.

**Comment by ankon** [comment]
Note that the error only happens when actually writing the output, not at the time you call `addSlide`. 

I agree that it would be useful to know _which_ slide had the issue, but that seems like a lot of error checking that would need to be added. 

The TypeScript definitions define that `text` must not be null/undefined, so if you use TypeScript *and* have proper types everywhere in your app there's a good chance that you'll be able to catch the error at development time rather than at runtime.

**Comment by gitbrent** [comment]
This is fixed thanks to Pull #695
